pinot-server/src/main/java/org/apache/pinot/server/api/resources/SegmentMetadataFetcher.java

```diff
@@ -62,24 +76,63 @@ public static String getSegmentMetadata(SegmentDataManager segmentDataManager, L
       columnSet = new HashSet<>(columns);
     }
     ObjectNode segmentMetadataJson = (ObjectNode) segmentMetadata.toJson(columnSet);
-    segmentMetadataJson.set("indexes", JsonUtils.objectToJsonNode(getIndexesForSegmentColumns(segmentDataManager)));
+    segmentMetadataJson.set(COLUMN_INDEX_KEY, JsonUtils.objectToJsonNode(getIndexesForSegmentColumns(segmentDataManager)));
+    segmentMetadataJson.set(STAR_TREE_INDEX_KEY, JsonUtils.objectToJsonNode((getStartreeIndexForSegmentColumns(segmentDataManager))));
     return JsonUtils.objectToString(segmentMetadataJson);
   }
 
+  /**
+   * Get the JSON object containing star tree index details for a segment.
+   */
+  private static List<Map<String, Object>> getStartreeIndexForSegmentColumns(SegmentDataManager segmentDataManager) {
+    List<Map<String, Object>> startreeDetails = new ArrayList<>();
+    ImmutableSegmentDataManager immutableSegmentDataManager = (ImmutableSegmentDataManager) segmentDataManager;
+    ImmutableSegment immutableSegment = immutableSegmentDataManager.getSegment();
+    List<StarTreeV2> starTrees = immutableSegment.getStarTrees();
+    if (starTrees == null) {
+      return startreeDetails;
+    }
+    startreeDetails = getImmutableSegmentStartreeIndexes(starTrees);
+    return startreeDetails;
```

Comment:
Can be simplified. Also suggest returning `null` if index does not exist
```suggestion
    List<StarTreeV2> starTrees = segmentDataManager.getSegment().getStarTrees();
    return starTrees != null ? getImmutableSegmentStartreeIndexes(starTrees) : null;
```

Reply:
This is done